### PR TITLE
docs: add HISTORY section to openssl-fipsinstall

### PR DESCRIPTION
**Pull request by xnox**
Documents when the command was added (3.0.0?) and when various options
were added after that.

Note sure about formatting, looks ok as a manpage.

### CODE REVIEW

**Review by paulidale — COMMENTED** [review]
Would sorting the options alphabetically be useful?

**Comment by xnox** [comment]
> Would sorting the options alphabetically be useful?

they are listed in order they appear in to the top level options list; same order as the long descriptions are; filtered for when added. No idea if that is useful ordering, and if we should reorder things in the top level options list / long description too.

File: doc/man1/openssl-fipsinstall.pod.in
Comment (slontis) [inline doc/man1/openssl-fipsinstall.pod.in:466]: x942 also?

File: doc/man1/openssl-fipsinstall.pod.in
Comment (levitte) [inline doc/man1/openssl-fipsinstall.pod.in:449]:
```suggestion
B<-ems_check>,
B<-self_test_oninstall>
```

File: doc/man1/openssl-fipsinstall.pod.in
Comment (levitte) [inline doc/man1/openssl-fipsinstall.pod.in:454]:
```suggestion
B<-pedantic>,
B<-no_drbg_truncated_digests>
```

File: doc/man1/openssl-fipsinstall.pod.in
Comment (levitte) [inline doc/man1/openssl-fipsinstall.pod.in:481]:
```suggestion
B<-hmac_key_check>,
B<-kmac_key_check>,
B<-signature_digest_check>,
B<-hkdf_digest_check>,
B<-tls13_kdf_digest_check>,
B<-tls1_prf_digest_check>,
B<-sshkdf_digest_check>,
B<-sskdf_digest_check>,
B<-x963kdf_digest_check>,
B<-dsa_sign_disabled>,
B<-no_pbkdf2_lower_bound_check>,
B<-no_short_mac>,
B<-tdes_encrypt_disabled>,
B<-rsa_pkcs15_padding_disabled>,
B<-rsa_pss_saltlen_check>,
B<-rsa_sign_x931_disabled>,
B<-hkdf_key_check>,
B<-kbkdf_key_check>,
B<-tls13_kdf_key_check>,
B<-tls1_prf_key_check>,
B<-sshkdf_key_check>,
B<-sskdf_key_check>,
B<-x963kdf_key_check>,
B<-ecdh_cofactor_check>
```

**Review by levitte — COMMENTED** [review]
Because POD works with paragraphs, the list of options will become one rather than a vertical list.  Therefore, commas between them to enhance the reading experience.

File: doc/man1/openssl-fipsinstall.pod.in
Comment (xnox) [reply doc/man1/openssl-fipsinstall.pod.in:466]: once it gets merged, yes!

File: doc/man1/openssl-fipsinstall.pod.in
Comment (paulidale) [reply doc/man1/openssl-fipsinstall.pod.in:481]: I agree with @levitte.  Either add commas or make a formal list of these.
My preference is for the former.

**Comment by xnox** [comment]
With `man --nj -l ./doc/man/man1/openssl-fipsinstall.1`

```
HISTORY
       The openssl-fipsinstall application was added in OpenSSL 3.0.

       The following options were added in OpenSSL 3.1:

       -ems_check, -self_test_oninstall

       The following options were added in OpenSSL 3.2:

       -pedantic, -no_drbg_truncated_digests

       The following options were added in OpenSSL 3.4:

       -hmac_key_check, -kmac_key_check, -signature_digest_check,
       -hkdf_digest_check, -tls13_kdf_digest_check, -tls1_prf_digest_check,
       -sshkdf_digest_check, -sskdf_digest_check, -x963kdf_digest_check,
       -dsa_sign_disabled, -no_pbkdf2_lower_bound_check, -no_short_mac,
       -tdes_encrypt_disabled, -rsa_pkcs15_padding_disabled,
       -rsa_pss_saltlen_check, -rsa_sign_x931_disabled, -hkdf_key_check,
       -kbkdf_key_check, -tls13_kdf_key_check, -tls1_prf_key_check,
       -sshkdf_key_check, -sskdf_key_check, -x963kdf_key_check,
       -x942kdf_key_check, -ecdh_cofactor_check
```

File: doc/man1/openssl-fipsinstall.pod.in
Comment (xnox) [reply doc/man1/openssl-fipsinstall.pod.in]: added, ahead of time.

**Review by slontis — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Suggest this goes to 3.4 given the large number of options added in that version.
If others disagree, I won't press for this.

**Comment by slontis** [comment]
Ideally it would go into 3.4

**Comment by t8m** [comment]
This is a doc update -> should go to all branches where it applies.

**Review by shahsb — APPROVED** [review]

**Review by levitte — APPROVED** [review]

**Comment by levitte** [comment]
> This is a doc update -> should go to all branches where it applies.

Yup. Whoever that cherry-picks it will have to take a bit of care to get things right, though

**Comment by xnox** [comment]
Would it help for me to split it into commit per version? (Although merges squash things, no?!)

**Comment by paulidale** [comment]
Per version commits would be ideal.

**Comment by xnox** [comment]
Split into individual commits by version.

**Review by paulidale — APPROVED** [review]

**Comment by levitte** [comment]
Having verified that other than splitting into nice commits, nothing has changed since the first set of approvals, I'll go ahead and merge this now.

**Comment by t8m** [comment]
Merged (selectively) to all active branches. Thank you for your contribution and the nicely prepared commits for cherry-picking.

**Pull request closed without merging**
